__tests__/supertest.js

```diff
@@ -0,0 +1,417 @@
+/*
+ * The integration test to test the whole microservice, with DB
+ */
+require('dotenv').config();
+const request = require('supertest');
+const server = require('../server/app');
+const { expect } = require('chai');
+const { RequestObject, ...seed } = require('../seed');
+const log = require('loglevel');
+const sinon = require('sinon');
+const { v4: uuidv4 } = require('uuid');
+
+describe('microservice integration tests', () => {
+  // beforeEach(async () => {
+  //   // before all, seed data to DB
+  //   await seed.seed();
+  //   // do any other setup here
+  //   // including authorize to the service if required
+  // });
+
+  // afterEach((done) => {
+  //   after finished all the test, clear data from DB
+  //   seed.clear().then(() => {
+  //     //In case other sinon stub would affect me
+  //     sinon.restore();
+  //     done();
+  //   });
+  // });
+
+  describe('E2E tests for /raw-captures end point', () => {
+    it(`Should raise validation error with error code 422 -- uuid is required `, function (done) {
+      const request_object = new RequestObject();
+      request_object.delete_property('uuid');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- uuid is not valid `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('uuid', 'sdf');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- image_url is required `, function (done) {
```

Comment:
We could also check the error message returned to see if it contains the Validation Error. I have a set of changes I am working on right now and testing that would return a concise error message. For now this is good, we will revisit it.

Reply:
Okay